### ISSUE TITLE
Bug: [no-unnecessary-condition] rule message is incorrect for multiple negations

### ISSUE THREAD

**Issue by abrahamguo** [typescript-eslint/typescript-eslint#9939]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.ts&code=MYewdgzgLgBAhjAvDKAnArgUwNwCgCWAZjABRwCUMA3jAL4HEkCEF1dDpTLlNtQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const a = true;
if (a) { } // Unnecessary conditional, value is always truthy. ✅
if (!a) { } // Unnecessary conditional, value is always falsy. ✅
if (!!a) { } // Unnecessary conditional, value is always falsy. ❌
```

### ESLint Config

```javascript
module.exports = {
  rules: {
    "@typescript-eslint/no-unnecessary-condition": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strict": true
  }
}
```

### Expected Result

The message should accurately reflect repeated negations.

### Actual Result

The message does not consider repeated negations.

**Comment by kirkwaiblinger** [comment]
A thought - I don't find it clear what "value" even refers to here... In `if (!a)`, is it `a` or `!a`? (Note that just `a` is underlined in all 3 cases, which would make the `!a` case seem incorrect and the `!!a` case seem correct to me). Maybe in addressing this, we could find a way to make that clear?

**Comment by Josh-Cena** [comment]
This rule only flags conditions, so surely the "value" is referring to whatever is being tested as a condition?

**Comment by kirkwaiblinger** [comment]
> This rule only flags conditions, so surely the "value" is referring to whatever is being tested as a condition?

Suppose you have a situation the following though, where we have multiple values and multiple things being tested as a condition. 
```ts
if (!a || Math.random() > 0.5) { }
//   ~ Unnecessary conditional, value is always falsy
```

What precedes it? `const a = true;` or `const a = false;`? 

To me, since `a` is underlined, it should be `const a = false;` but it is actually `const a = true;`.

If we changed it to be 
```ts
if (!a || Math.random() > 0.5) { }
//  ~~ Unnecessary conditional, value is always falsy
```
I think everyone would agree that it's `const a = true;` rather than there being two differing interpretations.

**Comment by kirkwaiblinger** [comment]
@abrahamguo 

Playing around with your PR (https://github.com/typescript-eslint/typescript-eslint/pull/9940), I definitely support the change to the  reporting location, and I'd also agree with the suggested wording change to "expression".

@Josh-Cena any objection to going forward with both those changes?

**Comment by Josh-Cena** [comment]
I also agree to expand the range. However I would keep it as "value"; only "value" can be "truthy" or "falsy". An "expression" is a source construct that _evaluates_ to a value.
